File: pyg_parent/pyg_manager/pyg_manager_interface/src/main/java/com/pinyougou/manager/service/GoodsService.java
package com.pinyougou.manager.service;
import java.util.List;
import com.pinyougou.pojo.TbGoods;

import entity.GoodsEditBean;
import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {


    /**
     * 保存商品的方法
     * @param goods
     */
    void saveGoods(GoodsEditBean goods);

    /**
     * 分页查询商品
     * @param pageNum
     * @param pageSize
     * @param goods
     * @return
     */
    PageResult<TbGoods> searchByGoods(int pageNum, int pageSize, TbGoods goods);

    /**
     * 修改商品状态
     * @param auditStatus
     * @param selectedIds
     */
    void updateAuditStatus(String auditStatus, List<Long> selectedIds);


    void updateIsMarketable(String isMarketable, List<Long> selectedIds);
}

File: pyg_parent/pyg_manager_web/src/main/webapp/admin/js/service/goodsService.js
//服务层
app.service('goodsService', function ($http) {

    this.updateAuditStatus = function (auditStatus, selectedIds) {
        return $http.get('../goods/updateAuditStatus/' + auditStatus + "/" + selectedIds);
    };

    //搜索
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../goods/search/' + pageNum + "/" + pageSize, searchEntity);
    }


    this.updateIsMarketable = function (isMarketable, selectIds) {
        return $http.get('../goods/updateIsMarketable/' + isMarketable + "/" + selectIds);
    }
});
